docs: response caching (#8345)
apollographql/router: docs/source/routing/performance/query-batching.mdx

- There are limitations on the ability of the router to preserve batches from the client request into the subgraph requests. In particular, certain forms of queries will require data to be present before they are processed. Consequently, the router will only be able to generate batches from queries which are processed which don't contain such constraints. This may result in the router issuing multiple batches or requests. 

> - If [query deduplication](/router/configuration/traffic-shaping/#query-deduplication), [response caching](/router/configuration/response-caching), or [entity caching](/router/configuration/entity-caching) are enabled, they will not apply to batched queries. Batching will take precedence over these features. Query deduplication, response caching, and entity caching will still be performed for non-batched queries.

Reviewer: ```suggestion
- If [query deduplication](/router/configuration/traffic-shaping/#query-deduplication), [response caching](/router/configuration/response-caching), or [entity caching](/router/configuration/entity-caching) are enabled, they do not apply to batched queries. Batching takes precedence over these features. These features still apply to non-batched queries.
```
